src/sentry/migrations/0310_backfill_slack_settings.py

```diff
@@ -0,0 +1,210 @@
+    ISSUE_ALERTS = 20
+
+    # Notifications for changes in assignment, resolution, comments, etc.
+    WORKFLOW = 30
+    # Notification when an issue happens shortly after your release.
+    ACTIVE_RELEASE = 31
+
+    # Notifications that require approval like a request to invite a member
+    APPROVAL = 40
+
+    # Notifications about quotas
+    QUOTA = 50
+
+    # Sub category of quotas for each event category
+    QUOTA_ERRORS = 51
+    QUOTA_TRANSACTIONS = 52
+    QUOTA_ATTACHMENTS = 53
+
+    # Sub category of quotas for warnings before hitting the actual limit
+    QUOTA_WARNINGS = 54
+
+
+class NotificationScopeType(Enum):
+    USER = 0
+    ORGANIZATION = 10
+    PROJECT = 20
+    TEAM = 30
+
+
+# these settings were defaulted to off for Slack
+SETTINGS_TO_BACKFILL = [
+    NotificationSettingTypes.DEPLOY,
+    NotificationSettingTypes.ISSUE_ALERTS,
+    NotificationSettingTypes.ACTIVE_RELEASE,
+    NotificationSettingTypes.WORKFLOW,
+]
+
+
+# backfill all entities for a given actor that's either a team or a user
+def backfill_one(apps, item_with_actor):
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    Team = apps.get_model("sentry", "Team")
+    # backfill to the scope of the entity
+    scope_type = (
+        NotificationScopeType.TEAM
+        if isinstance(item_with_actor, Team)
+        else NotificationScopeType.USER
+    )
+    for notification_type in SETTINGS_TO_BACKFILL:
+        # find the setting for that user/teams
+        settings = NotificationSetting.objects.filter(
+            scope_type=scope_type.value,
+            target_id=item_with_actor.actor_id,
+            provider=ExternalProviders.SLACK.value,
+            type=notification_type.value,
+            scope_identifier=item_with_actor.id,
+        )
+        setting = settings.first()
+        # if we have a setting that is not the default we can skip
+        if setting and setting.value != NotificationSettingOptionValues.DEFAULT.value:
+            continue
+
+        # if we have a row it's set the default which we need to change
+        if setting:
+            setting.value = NotificationSettingOptionValues.NEVER.value
+            setting.save()
+        else:
+            # otherwise create the backfill
+            NotificationSetting.objects.create(
+                scope_type=scope_type.value,
+                target_id=item_with_actor.actor_id,
+                provider=ExternalProviders.SLACK.value,
+                type=notification_type.value,
+                scope_identifier=item_with_actor.id,
+                value=NotificationSettingOptionValues.NEVER.value,
+            )
+
+
+# backfill all users who joined after START_DATE_DEFAULT_SLACK_NOTIFICAITON
+def backfill_users(apps):
+    Identity = apps.get_model("sentry", "Identity")
+    IdentityProvider = apps.get_model("sentry", "IdentityProvider")
+
+    for identity_provider in RangeQuerySetWrapperWithProgressBar(IdentityProvider.objects.all()):
```

Comment:
Any reason why you're `all()` instead of `IdentityProvider.objects.filter(type='slack')`

Reply:
@barkbarkimashark Yes, we don't have an index on that column

Reply:
Hmm, interesting .. its more efficient to grab all the rows than to filter by a non-indexed column? 

Well, no matter, I was just curious as all. I don't think it matters a lot in this case.

Reply:
@barkbarkimashark Efficient isn't necessarily the right word here. It's gonna eat up more CPU and memory for the migration but it's going to be a lighter load on Postgres. But this is standard practice for these kind of migrations (filtering items in-memory to use a indexed query).